execution/execution-correctness/src/execution_correctness.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use consensus_types::block::Block;
+use executor_types::{Error, StateComputeResult};
+use libra_crypto::HashValue;
+use libra_types::{
+    contract_event::ContractEvent, ledger_info::LedgerInfoWithSignatures, transaction::Transaction,
+};
+
+/// Interface for ExecutionCorrectness.
+/// It is basically the same as BlockExecutor except some interfaces will return signature with result.
+pub trait ExecutionCorrectness: Send {
```

Comment:
this has the same function signatures as BlockExecutor, why create a separate trait?

Reply:
Not the same. Take a look on `execute_block`